COMPOUND_COMPONENTS.md

```diff
@@ -0,0 +1,334 @@
+  tabPanelProps: React.HTMLAttributes<HTMLElement>
+}
+
+interface TabsProps {
+  tabListProps: React.HTMLAttributes<HTMLElement>
+  items: Tab[]
+}
+```
+
+Conversely, compound components have a more verbose implementation and loose control over structure. This flexibility allows developers to have direct access to underlying elements which makes manipulating attributes, styles, and markup structure much more natural and intuitive. From the above example:
+
+```tsx
+<Tabs>
+  <Tabs.List>
+    <Tabs.Item>First Tab</Tabs.Item>
+    <Tabs.Item>Second Tab</Tabs.Item>
+  </Tabs.List>
+  <Tabs.Panel>First Tab Contents</Tabs.Panel>
+  <Tabs.Panel>Second Tab Contents</Tabs.Panel>
+</Tabs>
+```
+
+Extra HTML attributes can be added directly to the `Tabs.List`, `Tabs.Item`, or `Tabs.Panel` sub-components.
+
+The pattern provides an additional benefit as well: maintainability. An active client-side application is constantly changing over the course of its lifecycle. However, not all parts change at the same rate. The application-level business logic (authentication, authorization, data fetching, et al) likely remains fairly intact over time. The UI logic layer (checkout flow steps, modal logic, etc) will change more frequently as features evolves or are deprecated. The most frequent changes happen at the markup structure and styling level. Configurable components are great at meeting the needs of your application today, but are more difficult to update. Changing the markup will often require changes to the component's code which will require library updates. These library updates mean more UI logic and complexity or complete rewrite to support more use-cases.
+
+For these reasons, we much prefer the compound component pattern. It allows our components to adapt to your application's needs and evolve with it over time, often without changes to our code.
+
+## Models
+
+### What is a Model?
+
+ If a compound component was stripped of all its markup, attributes, and styling, what would remain is the model. The model is how we describe the state and behavior of the component. You could completely swap out the underlying elements, attributes, and styles, and the model would remain the same. The model is an object that is composed of two parts: `state` and `events`. The model's `state` describes the current snapshot in time of the component, and `events` are functions that act on that state (behaviors).
+ 
+ ### Why Models?
+ 
+ Advantages of models:
+ * A common API structure to group state and behavior of components
+ * Atomic responsibilities
+ * Composable and shareable functionality
+ 
+ We use React Hooks to return models. An empty model would look like this:
+ 
+ ```ts
+ const useEmptyModel = (config = {}) => {
+   const state = {}
+   const events = {}
+   
+   return { state, events }
+ }
+ ```
+
+A model has configuration for functionality including configuration of events. An event has two parts:
+* Guard
+* Action
```

Comment:
Explain more of the of the why of these concepts. Also consider listing actions first - also change "actions" to "callbacks"